frame/staking/src/lib.rs

```diff
@@ -1703,8 +1667,14 @@ decl_module! {
 			} else {
 				// This was the consequence of a partial unbond. just update the ledger and move on.
 				Self::update_ledger(&controller, &ledger);
-				// This is only an update, so we use less overall weight
-				Some(50 * WEIGHT_PER_MICROS + T::DbWeight::get().reads_writes(4, 2))
+
+				// This is only an update, so we use less overall weight.
+				// Note: we are re-calculating weights here which is not performant per se but it is
+				// accurate and maintainable. The calculation themselve are trivial and O(1).
+				Some(
+					T::WeightInfo::withdraw_unbonded_kill(num_slashing_spans)
+						.saturating_sub(T::WeightInfo::withdraw_unbonded_update(num_slashing_spans))
```

Comment:
I'm surprised to see a subtraction in weight calculations based on the generated weight calcs. Can you give your reasoning?

Reply:
So this is the case where we need to return the `T::WeightInfo::withdraw_unbonded_update`. I was wrong thinking that the `Option` that you return is the correction amount, while it is not like that. I need to just return `T::WeightInfo::withdraw_unbonded_update`.